src/Symfony/Bundle/FrameworkBundle/Test/BrowserKitAssertionsTrait.php

```diff
@@ -124,6 +126,21 @@ public static function assertRouteSame($expectedRoute, array $parameters = [], s
         self::assertThat(self::getRequest(), $constraint, $message);
     }
 
+    public static function assertThatForResponse(Constraint $constraint, string $message = ''): void
+    {
+        try {
+            self::assertThat(self::getResponse(), $constraint, $message);
+        } catch (ExpectationFailedException $exception) {
+            if (($serverExceptionMessage = self::getResponse()->headers->get('X-Debug-Exception'))
+                && ($serverExceptionFile = self::getResponse()->headers->get('X-Debug-Exception-File'))) {
+                $serverExceptionFile = explode(':', $serverExceptionFile);
+                $exception->__construct($exception->getMessage(), $exception->getComparisonFailure(), new \ErrorException(rawurldecode($serverExceptionMessage), 0, 1, rawurldecode($serverExceptionFile[0]), $serverExceptionFile[1]));
```

Comment:
I wonder if it's correct to override the previous exceptions and change the stacktrace order... at least we should pass the `$exception->getPrevious()` to the new `\ErrorException()` instance to keep the history, but I still think this new exception should be on top.

Reply:
I cannot be on top: that be a BC break, and that'd change the expectations ppl have when they pass the $message argument.
Also logically, this is the cirrect order: the previous is the nested one.

Reply:
or merged with actual.

Reply:
That'd also be a BC break.
I think the current code is what produces the best experience, see example in the description.
We're not building a design here, but a reporting experience.

Reply:
at least shouldn't we pass the `$exception->getPrevious()` to the new \ErrorException() instance to keep the previous stacktrace?

Reply:
I'm refering to the "Cause by" Stack trace, which is not in the current description `[...]`. not sure how it look.

Reply:
Oh, but this is a "remote" process - how would you forward the stack from the ErrorHandler component to this one? It looks to heavy for a header to me...